tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -246,128 +246,152 @@ export class ScalarCardContainer implements CardRenderer, OnInit {
+          if (xAxisType === XAxisType.RELATIVE && normalizedPoints.length) {
+            const firstPoint = normalizedPoints[0];
+            normalizedPoints = normalizedPoints.map((point) => ({
               ...point,
               x: point.x - firstPoint.x,
             }));
           }
-          return {id: runId, points};
+
+          return {seriesId: partial.seriesId, points: normalizedPoints};
         });
+      }),
+      // Smooth
+      combineLatestWith(this.store.select(getMetricsScalarSmoothing)),
+      switchMap<[PartialSeries[], number], Observable<ScalarCardDataSeries[]>>(
+        ([runsData, smoothing]) => {
+          const cleanedRunsData = runsData.map(({seriesId, points}) => ({
+            id: seriesId,
+            points,
+          }));
+          if (smoothing <= 0) {
+            return of(cleanedRunsData);
+          }
 
-        if (smoothing === 0) {
-          return of(dataSeriesList);
+          return from(classicSmoothing(cleanedRunsData, smoothing)).pipe(
+            map((smoothedDataSeriesList) => {
+              const smoothedList = cleanedRunsData.map((dataSeries, index) => {
+                return {
+                  id: getSmoothedSeriesId(dataSeries.id),
+                  points: smoothedDataSeriesList[index].points.map(
+                    ({y}, pointIndex) => {
+                      return {...dataSeries.points[pointIndex], y};
+                    }
+                  ),
+                };
+              });
+              return [...cleanedRunsData, ...smoothedList];
+            })
+          );
         }
-
-        return from(classicSmoothing(dataSeriesList, smoothing)).pipe(
-          map((smoothedDataSeriesList) => {
-            const smoothedList = dataSeriesList.map((dataSeries, index) => {
-              return {
-                id: getSmoothedSeriesId(dataSeries.id),
-                points: smoothedDataSeriesList[index].points.map(
-                  ({y}, pointIndex) => {
-                    return {...dataSeries.points[pointIndex], y};
-                  }
-                ),
-              };
-            });
-            return [...dataSeriesList, ...smoothedList];
-          })
-        );
-      }),
-      startWith([]),
-      shareReplay(1)
+      ),
+      startWith([] as ScalarCardDataSeries[])
     );
 
-    this.chartMetadataMap$ = runIdAndPoints$.pipe(
-      switchMap((runIdAndPoints) => {
-        if (!runIdAndPoints.length) {
+    this.chartMetadataMap$ = nonNullRunsToScalarSeries$.pipe(
+      switchMap<
+        RunToSeries<PluginType.SCALARS>,
+        Observable<Array<{seriesId: string; displayName: string}>>
+      >((runToSeries) => {
+        const runIds = Object.keys(runToSeries);
+        if (!runIds.length) {
           return of([]);
         }
 
         return combineLatest(
-          runIdAndPoints.map((runIdAndPoint) => {
-            return this.getRunDisplayNameAndPoints(runIdAndPoint);
+          runIds.map((seriesId) => {
```

Comment:
nit: same as above, might be clearer to see the runId to seriesId conversion if we have the parameter still be named `runId` and then make L392 `return {seriesId: runId, displayName}`.